plugins/flytekit-dgxc-lepton/flytekitplugins/dgxc_lepton/connector.py

```diff
@@ -0,0 +1,749 @@
+                log_info(f"Endpoint ready! State: {state}, URL: {external_endpoint}")
+
+                # Return both URL and endpoint name for potential deletion
+                output_message = f"URL: {external_endpoint} | Endpoint: {deployment_name}"
+                return Resource(
+                    phase=flyte_phase,
+                    message=f"Endpoint status: {state}",
+                    log_links=log_links,
+                    outputs={"o0": output_message},
+                )
+            else:
+                log_info(f"Endpoint not ready yet. State: {state}, continuing to poll")
+                return Resource(
+                    phase=flyte_phase,
+                    message=f"Endpoint status: {state}",
+                    log_links=log_links,
+                )
+
+        except (subprocess.TimeoutExpired, json.JSONDecodeError) as e:
+            log_info(f"Failed to get status for Lepton endpoint {deployment_name}: {e}")
+            return Resource(phase=convert_to_flyte_phase("failed"))
+        except Exception as e:
+            log_info(f"Unexpected error getting Lepton endpoint status {deployment_name}: {e}")
+            return Resource(phase=convert_to_flyte_phase("failed"))
+
+    async def delete(self, resource_meta: LeptonMetadata, **kwargs):
+        """Delete a Lepton endpoint using lep CLI"""
+
+        deployment_name = resource_meta.deployment_name
+
+        try:
+            log_info(f"Deleting Lepton endpoint: {deployment_name}")
+
+            # Delete endpoint using lep CLI
+            result = await asyncio.get_event_loop().run_in_executor(
+                None,
+                lambda: subprocess.run(
+                    ["lep", "endpoint", "remove", "--name", deployment_name],
+                    capture_output=True,
+                    text=True,
+                    timeout=60,
+                ),
+            )
+
+            if result.returncode == 0:
+                log_info(f"Successfully deleted Lepton endpoint: {deployment_name}")
+            else:
+                # Check if endpoint doesn't exist (idempotent deletion)
+                if "not found" in result.stderr.lower() or "404" in result.stderr:
+                    log_info(f"Endpoint {deployment_name} already deleted or doesn't exist - treating as success")
+                else:
+                    logger.error(f"Failed to delete Lepton endpoint: {result.stderr}")
+                    raise RuntimeError(f"Failed to delete Lepton endpoint: {result.stderr}")
+
+        except subprocess.TimeoutExpired:
+            logger.error(f"Timeout deleting Lepton endpoint: {deployment_name}")
+            raise RuntimeError(f"Timeout deleting Lepton endpoint: {deployment_name}")
+        except Exception as e:
+            logger.error(f"Error deleting Lepton endpoint: {e}")
+            raise
+
+    def get_logs(self, resource_meta: LeptonMetadata, **kwargs) -> GetTaskLogsResponse:
```

Comment:
where do you want to embed this? how about `message` in the `get` method? i think that does show up in the logs.